src/system/SystemLayerImplZephyr.cpp

```diff
@@ -0,0 +1,235 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements LayerImplZephyr.
+ */
+
+#include <lib/support/CodeUtils.h>
+#include <platform/LockTracker.h>
+#include <system/PlatformEventSupport.h>
+#include <system/SystemFaultInjection.h>
+#include <system/SystemLayer.h>
+#include <system/SystemLayerImplZephyr.h>
+
+namespace chip {
+namespace System {
+
+LayerImplZephyr::LayerImplZephyr() {}
+
+CHIP_ERROR LayerImplZephyr::Init()
+{
+    VerifyOrReturnError(mLayerState.SetInitializing(), CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(mLayerState.SetInitialized(), CHIP_ERROR_INCORRECT_STATE);
+
+    return CHIP_NO_ERROR;
+}
+
+void LayerImplZephyr::Shutdown()
+{
+    mLayerState.ResetFromInitialized();
+}
+
+CHIP_ERROR LayerImplZephyr::StartTimer(Clock::Timeout delay, TimerCompleteCallback onComplete, void * appState)
+{
+    assertChipStackLockedByCurrentThread();
+
+    VerifyOrReturnError(mLayerState.IsInitialized(), CHIP_ERROR_INCORRECT_STATE);
+
+    CHIP_SYSTEM_FAULT_INJECT(FaultInjection::kFault_TimeoutImmediate, delay = System::Clock::kZero);
+
+    CancelTimer(onComplete, appState);
+
+    TimerList::Node * timer = mTimerPool.Create(*this, SystemClock().GetMonotonicTimestamp() + delay, onComplete, appState);
+    VerifyOrReturnError(timer != nullptr, CHIP_ERROR_NO_MEMORY);
+
+    if (mTimerList.Add(timer) == timer)
+    {
+        // The new timer is the earliest, so the time until the next event has probably changed.
+        TriggerPlatformTimerUpdate();
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR LayerImplZephyr::ExtendTimerTo(Clock::Timeout delay, TimerCompleteCallback onComplete, void * appState)
+{
+    VerifyOrReturnError(delay.count() > 0, CHIP_ERROR_INVALID_ARGUMENT);
+
+    assertChipStackLockedByCurrentThread();
+
+    Clock::Timeout remainingTime = mTimerList.GetRemainingTime(onComplete, appState);
+    if (remainingTime.count() < delay.count())
+    {
+        if (remainingTime == Clock::kZero)
```

Comment:
nit: I know this is copied but I wonder if this is needed - `StartTimer` calls `CancelTimer`, which seems to check `mExpiredTimers` as well...